sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -336,4 +336,62 @@ public UnsafeArrayData copy() {
     arrayCopy.pointTo(arrayDataCopy, Platform.BYTE_ARRAY_OFFSET, sizeInBytes);
     return arrayCopy;
   }
+
+  public int[] toPrimitiveIntArray() {
```

Comment:
It would be hard to tell the difference between `toPrimitiveIntArray` and `toIntArray` by name and signature because both returns primitive arrays. How about `toIntArrayUnchecked`? Please add JavaDoc to explain the difference.